Increasing kafka.js performance

aalykiot opened the pull request:
Refactoring based on this: https://kafka.js.org/docs/consuming#a-name-concurrent-processing-a-partition-aware-concurrency

helio-frota reviewed (approved):
LGTM to increase the App performance, since the demo uses 4 partitions 
https://github.com/nodeshift-starters/reactive-koffeeshop-demo/blob/12d70e42eb45f247b7c26af8720ed366a1a14da2/create-topics.sh#L5

And we are letting kafkaJS handle this https://github.com/nodeshift-starters/reactive-koffeeshop-demo/blob/main/barista-kafka/src/kafka-barista.js#L40 without specifying.

aalykiot commented:
@helio-frota FYI, the way we prepared the beverage with the `async/await` syntax also introduced some performance penalties. That's why I switched it to`.then`.

mhdawson reviewed (approved):
LGTM
